src/Language/PureScript/Infernal.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS -fno-warn-unused-binds #-}
+
+module Language.PureScript.Infernal
+  ( typeSearch
+  , typeSearch'
+  ) where
+
+import           Protolude
+
+import           Control.Monad.Writer
+import           Data.Aeson (decodeStrict)
+import qualified Data.ByteString as BS
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Language.PureScript.TypeChecker.Entailment as Entailment
+
+import qualified Language.PureScript.TypeChecker.Monad      as TC
+import           Language.PureScript.TypeChecker.Subsumption
+import           Language.PureScript.TypeChecker.Unify as P
+
+import Language.PureScript.AST as P
+import Language.PureScript.Crash
+import Control.Monad.Supply as P
+import Language.PureScript.Environment as P
+import Language.PureScript.Errors as P
+import Language.PureScript.Names as P
+import Language.PureScript.Externs as P
+import Language.PureScript.Types as P
+import Language.PureScript.TypeChecker.Skolems as Skolem
+import Language.PureScript.TypeChecker.Synonyms as P
+import Language.PureScript.ModuleDependencies as P
+
+-- REPL functions
+loadModules :: [Text] -> IO P.Environment
+loadModules = loadEnv . map (\m -> "./externs/" <> toS m <> ".json")
+
+loadEnv :: MonadIO m => [FilePath] -> m P.Environment
+loadEnv modules = do
+  mkEnv <$> traverse readExternFile modules
+  where
+    readExternFile :: MonadIO m => FilePath -> m P.ExternsFile
+    readExternFile fp = do
+      parseResult <- liftIO (decodeStrict <$> BS.readFile fp)
+      case parseResult of
+        Nothing -> internalError "parsing an extern failed"
+        Just externs -> pure externs
+
+-- Potentially needed for psc-ide integration
```

Comment:
Before this is merged, please take out the psc-ide specific code. We can hopefully move it under `Language.PureScript.Ide` later. Actually, while we're talking modules, this one should probably go under `Language.PureScript.Typechecker`